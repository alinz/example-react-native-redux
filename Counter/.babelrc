{
 "presets": [ "react-native-stage-0" ]
}