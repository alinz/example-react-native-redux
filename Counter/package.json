{
  "name": "Counter",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "react-native start",
    "test": "nyc mocha --opts ./app/tests/mocha.opts --recursive ./app/tests/*/*.js ./app/tests/*/*/*.js",
    "setup_project": "npm run step1; npm run step2",
    "step1": "rm -rf node_modules/react-native/node_modules/react; rm -rf node_modules/react-native/node_modules/fbjs; rm -rf node_modules/react/node_modules/fbjs",
    "step2": "cd node_modules; find . -name .babelrc | grep -v packager | xargs rm",
    "testjest": "jest",
    "testjest_update": "jest -u",
	"appium":"appium",
	"run-e2e":"wdio app/tests/wdio.conf.js"
  },
  "jest": {
    "preset": "jest-react-native"
  },
  "dependencies": {
    "react": "15.2.1",
    "react-native": "^0.31.0",
    "react-redux": "4.4.0",
    "redux": "3.3.1",
    "redux-thunk": "1.0.3"
  },
  "devDependencies": {
    "appium": "^1.4.3",
    "babel-core": "^6.14.0",
    "babel-jest": "^14.1.0",
    "babel-preset-react-native-stage-0": "^1.0.1",
    "chai": "^3.5.0",
    "enzyme": "^2.4.1",
    "jest": "^14.1.0",
    "jest-react-native": "^14.1.3",
    "nyc": "^8.1.0",
    "react-dom": "^15.3.1",
    "react-native": "^0.31.0",
    "react-native-mock": "^0.2.6",
    "react-test-renderer": "^15.3.1",
    "wdio": "^0.3.3",
    "wdio-mocha-framework": "^0.4.0",
    "wdio-spec-reporter": "0.0.3",
    "webdriverio": "^4.2.8"
  }
}
