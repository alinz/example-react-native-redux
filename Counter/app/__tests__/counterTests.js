/**
 * Created by rotem on 27/08/2016.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../components/counter';


describe('Counter', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <Counter/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with text', () => {
        const tree = renderer.create(
            <Button text="Up"/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});