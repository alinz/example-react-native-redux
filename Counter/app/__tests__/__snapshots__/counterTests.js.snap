exports[`Counter renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center"
    }
  }>
  <Text
    accessibilityLabel="Counter value text"
    accessible={true}
    testID="Counter value text" />
  <View
    accessibilityLabel="button Up"
    accessible={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "height": 30,
        "justifyContent": "center",
        "margin": 3,
        "padding": 10,
        "width": 100
      }
    }
    testID="button Up">
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Object {
          "opacity": 1
        }
      }
      testID={undefined}>
      <Text>
        Up
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="button Down"
    accessible={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "height": 30,
        "justifyContent": "center",
        "margin": 3,
        "padding": 10,
        "width": 100
      }
    }
    testID="button Down">
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Object {
          "opacity": 1
        }
      }
      testID={undefined}>
      <Text>
        Down
      </Text>
    </View>
  </View>
</View>
`;
